recipes/cpython/all/conanfile.py

```diff
@@ -170,42 +177,42 @@ def source(self):
     def _with_libffi(self):
         # cpython 3.7.x on MSVC uses an ancient libffi 2.00-beta (which is not available at cci, and is API/ABI incompatible with current 3.2+)
         return self._supports_modules \
-               and (self.settings.compiler != "Visual Studio" or tools.Version(self._version_number_only) >= "3.8")
+               and (not is_msvc(self) or Version(self._version_number_only) >= "3.8")
 
     def requirements(self):
-        self.requires("zlib/1.2.11")
+        self.requires("zlib/[>=1.2.11 <2]", force=True)
         if self._supports_modules:
-            self.requires("openssl/1.1.1l")
-            self.requires("expat/2.4.1")
+            self.requires("openssl/[>=1.1 <4]")
+            self.requires("expat/2.5.0")
             if self._with_libffi:
-                self.requires("libffi/3.2.1")
-            if tools.Version(self._version_number_only) < "3.8":
+                self.requires("libffi/3.4.4")
+            if Version(self._version_number_only) < "3.8":
                 self.requires("mpdecimal/2.4.2")
-            elif tools.Version(self._version_number_only) < "3.10":
+            elif Version(self._version_number_only) < "3.10":
                 self.requires("mpdecimal/2.5.0")
             else:
                 self.requires("mpdecimal/2.5.0")  # FIXME: no 2.5.1 to troubleshoot apple
         if self.settings.os != "Windows":
-            if not tools.is_apple_os(self.settings.os):
-                self.requires("libuuid/1.0.3")
-            self.requires("libxcrypt/4.4.25")
+            if not is_apple_os(self):
+                self.requires("util-linux-libuuid/2.39.2", force=True)
```

Comment:
Checking the PR now before merging, why is force needed here?

Reply:
It's getting a version conflict with another dependency that uses util-linux-libuuid, in 2.x this doesn't seem to force the version? Is there a better way to do this?

Reply:
@Ahajha which dependency is it conflicting with? I can't see anything in the logs - perhaps you need to update your local cache?

The normal resolution to this kind of issue is to bump the `util-linux-libuuid` version on the other dependency so that both are at the most recent version.

Reply:
Ah, looks like it was a stale cache. It came from `fontconfig`, which comes from `tk`. I'll remove it.